app/webroot/js/wjr/map.js

```diff
@@ -112,18 +116,20 @@ define(['jquery', 'underscore', 'async!https://maps.googleapis.com/maps/api/js?k
                 if (_.has(markers, entity.id)) {
                   return;
                 }
-                var marker, infoWindow;
+                var marker, color, infoWindow;
                 infoWindow = new google.maps.InfoWindow({
                   content: '<a href="' + entity.url +
                            '"><h3>' + entity.name + '</h3></a>'
                 });
+                color = entity.map_standard.color ? entity.map_standard.color : 'rgba(0,0,0,1)';
                 marker = new google.maps.Marker({
-                  position: new google.maps.LatLng(entity.lat, entity.lng)
+                  position: new google.maps.LatLng(entity.lat, entity.lng),
+                  icon:  fetchDomain + '/maps/icons/' + color + '.png',
```

Comment:
Instead of rendering the image server side in Rails and then using it here, can we make use of Google Maps SVG rendering support?

The main advantages I see of this:
- renders better on devices with different pixel densities (phones, some computer displays)
- simpler - no need to depend on Rails to render the image for us beforehand.

I tried it out and it seems to work pretty well (see docs here: https://developers.google.com/maps/documentation/javascript/symbols).
```
var icon = {
        path: "M12,11.5A2.5,2.5 0 0,1 9.5,9A2.5,2.5 0 0,1 12,6.5A2.5,2.5 0 0,1 14.5,9A2.5,2.5 0 0,1 12,11.5M12,2A7,7 0 0,0 5,9C5,14.25 12,22 12,22C12,22 19,14.25 19,9A7,7 0 0,0 12,2Z",
        fillColor: '#35b522',
        fillOpacity: 1.0,
        anchor: new google.maps.Point(0,0),
        strokeOpacity: 0.5,
        strokeWeight: 2.0,
        strokeColor: '#000000',
        scale: 1.5
      };
```

You'd just need to use the map_standard color in this.

What do you think?

Reply:
Great idea.  SVG support is much better than the system I implemented, I didn't realize that Google Maps supported it natively.  Done.